### ISSUE TITLE
s/coll-of on sets fails

### ISSUE THREAD

**Issue by athos** [bhb/expound#31]
```clj
=> (ex/expound (s/coll-of integer?) #{:a})
java.lang.UnsupportedOperationException: nth not supported on this type: PersistentHashSet
```

As far as I know, `s/coll-of` should be applicable to non-sequantial collections (including sets).
According to the stack trace, it looks like the error occurs here:

https://github.com/bhb/expound/blob/7853f0fb1da5acff0ac9e26fefd87dc0ce93405a/src/expound/paths.cljc#L81

**Comment by bhb** [comment]
Thanks very much for reporting this! I can reproduce this and I think I have a fix. Can you try 0.3.1-SNAPSHOT and see if that works for your project?

**Comment by athos** [comment]
Thank you for your quick reply! I see the things work fine. Thanks a lot 😄

**Comment by bhb** [comment]
Thanks again for your help on this! I've just released Expound 0.3.1 which includes this fix.
